modules/stream-shredder/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/shredder/stream/parquet/Codecs.scala

```diff
@@ -0,0 +1,61 @@
+package com.snowplowanalytics.snowplow.rdbloader.shredder.stream.parquet
+
+import com.github.mjakubowski84.parquet4s._
+import com.snowplowanalytics.snowplow.rdbloader.common.transformation.WideField
+import com.snowplowanalytics.snowplow.rdbloader.common.transformation.WideField.FieldValue
+
+import java.sql.Date
+import java.time.temporal.ChronoUnit
+import java.time.{Instant, ZoneOffset}
+import java.util.TimeZone
+import scala.util.Try
+
+object Codecs {
+
+  val config = ValueCodecConfiguration(TimeZone.getTimeZone(ZoneOffset.UTC))
+
+  implicit val cod: ValueCodec[FieldValue] = new ValueCodec[FieldValue] {
+
+    override def encode(data: FieldValue, configuration: ValueCodecConfiguration): Value = data match {
+      case FieldValue.NullValue => NullValue
+      case WideField.FieldValue.StringValue(value) =>
+        implicitly[ValueCodec[String]].encode(value, config)
+      case WideField.FieldValue.BooleanValue(value) =>
+        implicitly[ValueCodec[Boolean]].encode(value, config)
+      case WideField.FieldValue.IntValue(value) =>
+        implicitly[ValueCodec[Int]].encode(value, config)
+      case WideField.FieldValue.DoubleValue(value) =>
+        implicitly[ValueCodec[Double]].encode(value, config)
+      case WideField.FieldValue.TimestampValue(value) =>
+        implicitly[ValueCodec[Long]].encode(ChronoUnit.MICROS.between(Instant.EPOCH, value.toInstant), config)
```

Comment:
Currently this is not compatible with what the batch transformer does.  So we need to reconcile the two.

By default, Spark writes timestamps using Parquet's INT96 column type.  If you wanted the stream transformer to do the same then you would need to use [the sqlTimestampEncoder](https://github.com/mjakubowski84/parquet4s/blob/2e353c86a8c0a12ac844eaecc2033f987bb09971/core/src/main/scala/com/github/mjakubowski84/parquet4s/ValueCodecs.scala#L235-L238) here.

On the other hand [this thread talks about INT96 being deprecated and discouraged](https://issues.apache.org/jira/browse/PARQUET-323).  And I believe we can change the spark behaviour by setting `spark.conf.set("spark.sql.parquet.outputTimestampType", "TIMESTAMP_MICROS")`.

The important question is... what does Databricks do if it receives: a timestamp_micros type, vs a int96 type?  Is there any difference for downstream analytic queries?